modules/build/src/main/scala/scala/build/Build.scala

```diff
@@ -1029,7 +1029,8 @@ object Build {
     either {
 
       val options0 =
-        if (sources.hasJava && !sources.hasScala)
+        // FIXME: don't add Scala to pure Java test builds (need to add pure Java test runner)
+        if (sources.hasJava && !sources.hasScala && scope != Scope.Test)
```

Comment:
We should add a pure Java test runner at some points, so that we can keep Java test builds pure as well.

Reply:
Created an issue: 
- https://github.com/VirtusLab/scala-cli/issues/3010